app/models/health/careplan_saver.rb

```diff
@@ -22,57 +20,86 @@ def create
-            signature_date = @careplan.provider_signed_on
-            @qualifying_activity.date_of_activity = signature_date
-
-            case @careplan.provider_signature_mode.to_s
-            when 'email'
-              @qualifying_activity.mode_of_contact = :other
-              @qualifying_activity.reached_client = :collateral
-              @qualifying_activity.mode_of_contact_other = 'On-line'
-              @qualifying_activity.reached_client_collateral_contact = 'On-line Signature'
-            when 'in_person'
-              @qualifying_activity.mode_of_contact = :in_person
-              @qualifying_activity.reached_client = :yes
-            else
-              # default to email signature
-              @qualifying_activity.mode_of_contact = :other
-              @qualifying_activity.reached_client = :collateral
-              @qualifying_activity.mode_of_contact_other = 'On-line'
-              @qualifying_activity.reached_client_collateral_contact = 'On-line Signature'
-            end
+          if @careplan.just_finished? && @create_qa
+            qualifying_activity = setup_care_planning_qualifying_activity
+          elsif @careplan.just_signed? && @create_qa
+            qualifying_activity = setup_pctp_signed_qualifying_activity
           end
 
-          @careplan.save
+          @careplan.save!
 
-          if should_really_create_qa
-            @qualifying_activity.source_id = @careplan.id
+          if qualifying_activity.present?
+            qualifying_activity.source_id = @careplan.id
 
-            @qualifying_activity.save
+            qualifying_activity.save
           end
           @careplan.set_lock
         end
-      rescue Exception => e
+      rescue Exception
         success = false
       end
       return success
     end
 
-    protected def setup_qualifying_activity
+    protected def setup_care_planning_qualifying_activity
+      if @careplan.member_verbalizes_understanding?
+        mode_of_contact = :phone
+        text = 'This writer completed Care Plan with patient. Patient gave verbal approval to Care Plan due to COVID-19.'
+      else
+        mode_of_contact = :in_person
+        text = 'This writer completed Care Plan with patient. Patient agreed to care plan.'
+      end
+
+      Health::QualifyingActivity.new(
+        source_type: @careplan.class.name,
+        user_id: @user.id,
+        user_full_name: @user.name_with_email,
+        activity: :care_planning,
+        date_of_activity: @careplan.patient_signed_on,
+        mode_of_contact: mode_of_contact,
+        reached_client: :yes,
+        follow_up: text,
+        patient_id: @careplan.patient_id,
+      )
+    end
+
+    protected def setup_pctp_signed_qualifying_activity
+      case @careplan.provider_signature_mode.to_s
+      when 'email'
+        mode_of_contact = :other
+        reached_client = :collateral
+        mode_of_contact_other = 'On-line'
+        reached_client_collateral_contact = 'On-line Signature'
```

Comment:
If the options are 'in_person' and 'everything else' seems like we probably don't need the email case.

Reply:
This is the pctp_signed QA, which already existed, and reflects that mode of contact is for the PCP, no changes in that were requested.